### PR TITLE
Update documentation to match the spec.

### PR DESCRIPTION
**Pull request by iphydf**

### CODE REVIEW

File: src/tox/Network/Tox.lhs [hunk 1]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
 100]}).  Open ranges are denoted \texttt{[n,]} to mean a minimum length of
 \texttt{n} with no specified maximum length.
 
+\section{TODO: Goals and threat model}
+
+This section should give an idea on what are the goals and non-goals of Tox, so
+that reader
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: can you finish this thought?

File: src/tox/Network/Tox.lhs [hunk 2]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
...
+(TODO: this is just a placeholder; some more technical description should be
+given)
+
+What Tox Does:
+
+\begin{itemize}
+  \item Authentication: user's address is its public key, thus "adding friend"
+    actually means verifying key (false in case of toxme?); drawback is that
+    after being compromised you have to generate absolutely new identity; also
+    it complicates working with multiple devices; (TODO: how about some web of
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: wob is a mistake, and if this is supposed to be readable as the spec, shouldn't speculation/ideas/planning be moved elsewhere?

File: src/tox/Network/Tox.lhs [hunk 3]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
...
+\end{itemize}
+
+(TODO: this is just a placeholder; some more technical description should be
+given)
+
+What Tox Does:
+
+\begin{itemize}
+  \item Authentication: user's address is its public key, thus "adding friend"
+    actually means verifying key (false in case of toxme?); drawback is that
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: While it is false in the case of toxme, that's not really relevant to tox. If I start a website that saves passwords but I only support http, that doesn't' make [name of popular web service] less secure.

File: src/tox/Network/Tox.lhs [hunk 4]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
...
+    after being compromised you have to generate absolutely new identity; also
+    it complicates working with multiple devices; (TODO: how about some web of
+    trust, master keys and subkeys here?)
+
+  \item End-to-end encryption: all the messages are encrypted via keys derived via
+    DH, thus keys are only known to sender and receiver and are never
+    transfered over network
+
+  \item Forward secrecy (?): can be achieved by using ephemeral keys (TODO: how are
+    they used in the current protocol? is the problem actually solved?)
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: somewhat: The specific use case that the original author planned to use toxcore personally does effectively guarantee forward secrecy. While in actuality, there's no such promise made, merely; it'll probably work.

File: src/tox/Network/Tox.lhs [hunk 5]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
...
+
+  \item Reliability:
+    \begin{itemize}
+      \item Tox is supposed to be fully decentralized network which doesn't depend
+        on any Single Point of Failure; this is achieved by using DHT, though
+        you still need an entry point;
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: wartoxing would solve this issue

File: src/tox/Network/Tox.lhs [hunk 6]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
...
+      \item Tox is supposed to work under (almost) any kind of NAT and firewall;
+        this is achieved by using hole-punching, UPnP and TCP-relays;
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: IIRC UPnP isn't merged into upstream master.

File: src/tox/Network/Tox.lhs [hunk 7]
```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
...
+    \end{itemize}
+
+  \item (Near-)Zero-conf: end-user should be able to \emph{just} use the
+    messenger;
+\end{itemize}
+
+What Tox Does NOT:
+
+\begin{itemize}
+  \item Tox does not care about anonymity; Unless TCP mode is used, participants
```
Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: `does not promise anonymity, but offers reasonable options for external tools `

File: src/tox/Network/Tox.lhs (hunk 3 above)
Comment (aaannndddyyy) [inline src/tox/Network/Tox.lhs]: Here I disagree. If a website uses super secure backends to store the data, but has doors wide open in the process of information transfer, then I would not call this service/website secure. I could call components of it secure. So it should still be mentioned prominently in various places, that using toxme is not providing the same level of security as using a tox id.

Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: @aaannndddyyy where would you draw the line? Should you also mention the issues with TOR, and with HTTPS in general, or just the toxme implementation? How about when you roll your own toxme? 

My point was, if you're trying to be fancy with 3rd party extensions, that decrease security at the cost of convenience, I don't think that's something we should concern ourselves with. Explaining it to client developers, sure, but the spec isn't for client devs.

(I've also updated my original comment with and additional line to make my original comment more clear.

Comment (aaannndddyyy) [inline src/tox/Network/Tox.lhs]: re drawing the line: make clear that basic working of tox is secure (tox id, etc.) but that if you rely on any third party-service, toxme being just an expample - it does not give the same security guarantees. I know, this is not about a website advertizing tox, and that it should be clear for a developer, but better be clear than face criticism for being insecure. All ways to use a service count, and what at the end matters is the user's security. User does not care if part xyz is what he used was secure, when his data leaked it leake, no matter which part leaked it. So anything that is supported should either be save or have a warning about potentially bing unsave. If a provider of any webservice says they hash all passwords, but receive them in plain text via http, then this service cannot claim to be secure. The line there would be that the transfer is part of the service. But if the user writes down his password on a noted next to his computer, this is not part of the service, and this insecurity is caused by the user.
That's how I see it.

Comment (GrayHatter) [inline src/tox/Network/Tox.lhs]: I don't disagree with your premise, only the location of the notice. Either way I assume @iphydf will read all of this, and ignore it. I was told that just a review for grammer is needed, as the content will be fixed up in later revisions.

Comment (aaannndddyyy) [inline src/tox/Network/Tox.lhs]: hehe, ok. And "grammar" ;)

**Pull request merged**